### ISSUE TITLE
[regression/net7.0] ISafeAreaView.IgnoreSafeArea isn't respected in .NET 7

### ISSUE THREAD

**Issue by AlexeyStarkov** [dotnet/maui#12823]
### Description

I've upgraded my project from .NET6 to .NET7 and noticed that IgnoreSafeArea I use within page content layout (i.e. Grid) is not respected anymore. SafeArea is still there whether IgnoreSafeArea is true or false. It works well in .NET6.

### Steps to Reproduce

1. Create .NET 7 MAUI project
2. Create ContentPage with Grid set as Content
3. Set Grid BackgroundColor to Grey and IgnoreSafeArea to True
4. Assign the newly created page to the MainPage property of App class
5. Run iOS application 

### Link to public reproduction project repository

https://github.com/AlexeyStarkov/MAUI_NET7_IgnoreSafeArea_Issue

### Version with bug

7.0 (current)

### Last version that worked well

6.0.424

### Affected platforms

iOS

### Affected platform versions

iOS 16.2

**Comment by bcaceiro** [comment]
I am facing the same problem.

**Comment by scriptBoris** [comment]
Any updates?

**Comment by anupkrjee** [comment]
I am facing same issue in .NET 7

**Comment by nathanjeynes** [comment]
This is an issue for us as well, this is basic functionality which has been broken in .NET 7 and should be fixed

**Comment by RobTF** [comment]
Yep, got this issue as well. Idea of ETA for fix would be much appreciated is basic required functionality.

**Comment by ianbaddino** [comment]
I also have this blocking issue. Is this working in .NET 6 / 8?

**Comment by mattleibow** [comment]
I just tested with .NET MAUI 7.0.81 using the iOS platform specific ad it seems to work when you set it on the `Page` not the layouts. The docs also indicate this: https://learn.microsoft.com/dotnet/maui/ios/platform-specifics/page-safe-area-layout

Looking at the code for the Layout, not sure that does anything as seems to only be half implemented as is not even a bindable property.

**Comment by RobTF** [comment]
Using it on the page is how I tried with no effect - to be honest I don't really see the point of adding the option onto individual layouts, as it should be a page-level concern anyway, shouldn't it? Xamarin only had the option on the page and worked fine.

The only thing that might be nice is to allow the developer to opt in/out of individual safe area directions (i.e. flow the page into the top, but not the bottom safe zone), but let's not run before we can walk (or crawl!).

I managed to work around this in the end with the following, painful hack;

Firstly, you must set the UseSafeArea to be false on the page.
```xml
<ContentPage xmlns="http://schemas.microsoft.com/dotnet/2021/maui"
             xmlns:x="http://schemas.microsoft.com/winfx/2009/xaml"
             xmlns:ios="clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;assembly=Microsoft.Maui.Controls"
             x:Class="MyApp.MyClass"
             ios:Page.UseSafeArea="False"
             Shell.NavBarIsVisible="False">
```

Then, create some service to fix-up the safe area insets.
```csharp
internal sealed class SafeAreaInsets : ISafeAreaInsets
{
    /// <inheritdoc/>
    public Thickness GetSafeAreaInsets()
    {
        if (!OperatingSystem.IsIOSVersionAtLeast(11))
        {
            return Thickness.Zero;
        }

        var window = UIApplication.SharedApplication.Delegate.GetWindow();
        var insets = window.SafeAreaInsets;
        return new Thickness(insets.Left, insets.Top, insets.Right, insets.Bottom);
    }
}
```

Then in the page...

```csharp
    /// <inheritdoc />
    protected override void OnAppearing()
    {
        base.OnAppearing();

#if __IOS__
        var services = IPlatformApplication.Current.GetProvider();
        var actualInsets = services.GetRequiredService<ISafeAreaInsets>().GetSafeAreaInsets();
        var topArea = actualInsets.Top;
        var bottomArea = actualInsets.Bottom;
        var safeInsets = On<iOS>().SafeAreaInsets();
        safeInsets.Top = -topArea;
        safeInsets.Bottom = -bottomArea;

        Padding = safeInsets;
#endif
    }
```

I found this here;
https://stackoverflow.com/questions/74724489/net-maui-ios-usesafearea-not-working-stacklayout-verticalstacklayout-and-grid

Basically, you pull the insets manually from the underlying OS (away from the broken MAUI abstraction), patch it into the insets object provided by the broken MAUI bits, then set a negative page padding manually to "drag" it into the safe inset space.

Horrible hack I know, but those ex-Xamarin ~victims~ developers you'll be used to it.

**Comment by hartez** [comment]
At the moment on .NET 7, the default for Page.UseSafeArea is `true`; on .NET 6, it was `false`. So now the page is containing the Grid to the safe area, no matter what you set for IgnoreSafeArea on the Grid. You can work around this by telling the ContentPage that it's allowed outside the safe area:

```xaml
<ContentPage ...
             xmlns:ios="clr-namespace:Microsoft.Maui.Controls.PlatformConfiguration.iOSSpecific;assembly=Microsoft.Maui.Controls"
             ios:Page.UseSafeArea="False">
```

You'll also need to tell the content (the Grid in the original example) to ignore the safe area.

**Comment by hartez** [comment]
After discussing this, we're going to revert back to the .NET 6 default for iOS, and leave the new default for Catalyst.

**Comment by AlexeyStarkov** [comment]
> I just tested with .NET MAUI 7.0.81 using the iOS platform specific ad it seems to work when you set it on the `Page` not the layouts. The docs also indicate this: https://learn.microsoft.com/dotnet/maui/ios/platform-specifics/page-safe-area-layout
> 
> Looking at the code for the Layout, not sure that does anything as seems to only be half implemented as is not even a bindable property.

As I remember the API you mentioned is the old one well known since Xamarin.Forms. The ISafeAreaView.IgnoreSafeArea I pointed out is the new property that can be specified within any view that implements the ISafeAreaView interface.

**Comment by bcaceiro** [comment]
So will we be able in the future to apply to controls? Because that is much necessary

**Comment by borrmann** [comment]
IgnoreSafeArea also can't be set in a StyleSheet for all Grids/ Layouts as it doesn't seem to have the necessary getters and setters to do so. Is there some sort of workaround to set the default to ignoring the safe area for all pages and layouts?
